lib/openstack_query/server_manager.py

```diff
@@ -0,0 +1,167 @@
+from typing import List, Dict, Any, Union
+from structs.query.query_details import QueryDetails
+from structs.query.preset_details import PresetDetails
+
+from enums.query.query_presets import QueryPresets
+from enums.query.server.server_properties import ServerProperties
+
+from openstack_query.query_server import QueryServer
+import openstack_query.manager_utils as manager_utils
+
+
+def search_all_servers(
+    cloud_account: str, query_details: QueryDetails
+) -> Union[str, List[Any]]:
+    return manager_utils.run_query(
+        cloud_account,
+        None,
+        query_details.properties_to_select,
+        query_details.group_by,
+        query_details.sort_by,
+    )
+
+
+def search_servers_older_than(
+    cloud_account: str,
+    query_details: QueryDetails,
+    days: int,
+    hours: int = 0,
+    minutes: int = 0,
+    seconds: int = 0,
+):
+    return manager_utils.run_query(
+        cloud_account,
+        QueryServer(),
```

Comment:
Having seen this used a few times, I think that the `manager_utils` should have a class which takes the `cloud_account` and `QueryX` as part of the construction

As a user I might want to do something akin to:

```
def find_between_interval():

manager = Foo(cloud_account, QueryServer())
preset_older = PresetDetails(
            preset=QueryPresets.OLDER_THAN,
            prop=ServerProperties.SERVER_CREATION_DATE,
           # ...etc.
)

older = manager.run_query(preset_older, ....)

preset_newer = PresetDetails(
            preset=QueryPresets.NEWER_THAN,
            prop=ServerProperties.SERVER_CREATION_DATE,
           # ...etc.
)

newer = manager.run_query(preset_newer)

return set(older) - set(newer)
```

That way you separate query state from a connection state into two calls effectively

Reply:
refactored `manager_utils` into class `QueryManager`